cloudferrylib/utils/local_db.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) 2014 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+import contextlib
+import json
+import logging
+import random
+import sqlite3
+import threading
+import time
+
+LOG = logging.getLogger(__name__)
+SQLITE3_DATABASE_FILE = 'migration_data.db'
+
+
+class Rollback(Exception):
```

Comment:
After some thought I decided to remove this exception and Transaction.rollback method completely.